imgee/utils.py

```diff
@@ -144,26 +180,55 @@ def get_s3_folder(f=''):
     return f or ''
 
 
-def download_frm_s3(img_name):
+def exists_in_s3(thumb):
+    folder = get_s3_folder()
+    bucket = get_s3_bucket()
+    extn = thumb.stored_file.extn
+    if 'thumb_extn' in ALLOWED_MIMETYPES[thumb.stored_file.mimetype]:
+        extn = ALLOWED_MIMETYPES[thumb.stored_file.mimetype]['thumb_extn']
+    key = os.path.join(folder, thumb.name + extn)
+    # print("checking whether exists in s3: {}".format(key))
+    resp = bucket.get_key(key)
+    if not resp:
+        # print("does not exist")
+        return False
+    # print("exists")
+    return True
+
+
+def download_from_s3(img_name):
     local_path = path_for(img_name)
     if not os.path.exists(local_path):
         bucket = get_s3_bucket()
+        folder = get_s3_folder()
         k = Key(bucket)
-        k.key = img_name
+        k.key = os.path.join(folder, img_name)
         k.get_contents_to_filename(local_path)
     return local_path
 
 
-def not_in_deleteQ(imgs):
-    # filters out the images which are queued for deletion and returns the remaining
-    return [i for i in imgs if not i.is_queued_for_deletion()]
-
-
 # -- image details --
 
 def get_width_height(img_path):
+    name, extn = os.path.splitext(img_path)
     try:
-        o = check_output('identify -quiet -ping -format "%wx%h" {}'.format(img_path), shell=True)
+        o = '0x0'
+        if extn in ['.pdf', '.gif']:
+            o = check_output('identify -quiet -ping -format "%wx%h" {}[0]'.format(img_path), shell=True)
+        elif extn in ['.cdr']:
+            wo = check_output('inkscape -z -W {}'.format(img_path), shell=True)
+            ho = check_output('inkscape -z -H {}'.format(img_path), shell=True)
```

Comment:
I tried a whole bunch of stuff to use just one command, but no use. Let's stick with this for now.